### DISCUSSION TITLE
UK place names are not specific enough

### DISCUSSION THREAD

**Discussion by djjudas21** [photoprism/photoprism#2095] [Q&A] [answered]
Hi there. I'm a new PhotoPrism user and I'm checking out the functionality on some holiday photos I took last month to see if it meets my needs before I migrate my whole library.

I noticed that the place names are not consistent. Where I live, in the UK, the location appears in the "States" view simply as England. Whereas the photos I took on holiday in Switzerland seem to appear either as the state (Basel-Landschaft) or the city (Zurich), as I would expect.

![Screenshot from 2022-03-02 17-40-50](https://user-images.githubusercontent.com/604595/156417794-8ae7b2fc-0f57-4734-b261-8b65132cdda3.png)

I don't know how your location data is formatted, but would it be possible for you to tweak your backend when working with UK locations? For example, I would classify my current location as:

> Country: UK
> Subdivision: England
> State: Bristol
> City: Bristol

and another location I visit frequently:

> Country: UK
> Subdivision: England
> State: Warwickshire
> City: Nuneaton

Here, I don't really care about the "subdivision" part of the location (England, Scotland, Wales, Northern Ireland) and the "state" or even the "city" part of the location is much more useful. I think most UK residents would probably agree with me.

I guess the UK is different from most other countries here, whose classification would go straight from "country" to "state" without the subdivision.

Thanks :smile:

**Comment by lastzero** [comment]
According to Wikipedia, UK is a kingdom and England a country 🧐

**Reply by djjudas21** [reply]
"It's complicated" :laughing: The UK and England are both countries, but England is not a sovereign state while the UK is.

Within the UK, we never really mention the "inner" country names, all you need is the postal code. Internationally, you typically would not expect to see England in a shipping address, but UK. Never both.

For example, off the top of my head I found this [international shipping label for a photo lab](https://www.ilfordphoto.com/wp/wp-content/uploads/2017/08/WEB-120-HARMAN.pdf) and the address given is 

> Harman Lab Direct
> Ilford Way
> Mobberly
> Knutsford
> WA16 7JL
> UK

https://goo.gl/maps/u7CAaeiFsg5aG7fA8

The country is given as UK, the county/state is omitted (it should be Cheshire), the city is Knutsford and the village is Mobberly. If I entered a photo at this location into PhotoPrism, I would expect to see Knutsford or possibly Cheshire as the "State".

**Comment by graciousgrey** [comment, marked as answer]
Agreed, this is something we can improve in the future. Thanks for reporting!

We already thought about creating those location albums for cities or even districts as well in case there are enough photos.

Feel free to open an idea issue :)

**Reply by djjudas21** [reply]
Thanks @graciousgrey - will do :slightly_smiling_face:

**Comment by aaronrudkin** [comment]
I added a reply to the underlying issue to note that Ireland has similar issues here and to describe options for fixing it. Thanks so much for your time, it's appreciated it, and absolutely no rush on this.

**Reply by lastzero** [reply]
Thanks a lot! We can take a look at it the next time we update our location database.

**Comment by ilazgo** [comment]
I also added a repply to the issue that maybe it would fit better here. I copy the tl;dr:

In Spain the term "State" refers to Spain. We should look for a generic name for the subdivisions, or we should look for another structure per country.
